code/localization/localize.cpp

```diff
@@ -481,6 +482,35 @@ void parse_tstringstbl(const char *filename)
 	parse_stringstbl_common(filename, true);
 }
 
+struct xstr_delayed_order {
+	SCP_string& toFill;
+	const char* name;
+	int xstr;
+};
+
+static void lcl_delayed_xstr_internal(tl::optional<xstr_delayed_order> to_init) {
+	static SCP_vector<xstr_delayed_order> delayed_init;
```

Comment:
Just as a point of clarification before anyone asks:
``delayed_init`` is very explicitly a static function variable as making it a global variable runs into issues of the order in which static globals are initialized, and it not being guaranteed to be initialized before the (also static global) option variables need to request a delayed XSTR. As a static function variable however, it is always guaranteed to be initialized by the time this function is called.
Yes, it does require this weird calling convention of fill the optional to add a value, keep it empty to fulfill the delayed XSTRs, but since this is an exclusively hidden part of the API, with any user just seeing the ``lcl_delayed_xstr`` function, it should be fine.